src/stan/variational/families/normal_fullrank.hpp

```diff
@@ -49,12 +49,15 @@ namespace stan {
 
     public:
       // Constructors
+
+      // Constructs a zero-filled variational distribution
       explicit normal_fullrank(size_t dimension) :
         dimension_(dimension) {
         mu_     = Eigen::VectorXd::Zero(dimension_);
-        L_chol_  = Eigen::MatrixXd::Identity(dimension_, dimension_);
+        L_chol_ = Eigen::MatrixXd::Zero(dimension_, dimension_);
```

Comment:
You should pull out the Cholesky factor validation --- as is, it's cut and pasted into the mutator (setter) and constructor.  Yet another reason why mutators are evil.